lib/carto/renderer_js.js

```diff
@@ -235,6 +238,13 @@ CartoCSS.prototype = {
       var done = {};
       for(var i = 0; i < defs.length; ++i) {
         var def = defs[i];
+
+        def.toXML(parse_env, {});
+
+        if (this.options.strict && parse_env.errors.message) {
```

Comment:
True! But, wouldn't be better to perform the `def.toXML` when the option is provided?